script.js

```diff
@@ -1,4 +1,306 @@
+        outcomeMessage = `It's a draw! <br> <br> Since you are the recent winner, you are the ultimate winner of muk-jji-ppa! <br> <br> You can play again by clicking "submit".`;
+      } else if (mostRecentWinner == "computer") {
+        mostRecentWinner = ""; // to restart the game
+        outcomeMessage = `It's a draw! <br> <br> Since the computer is the recent winner, you have lost muk-jji-ppa! <br> <br> You can play again by clicking "submit".`;
+      } else if (mostRecentWinner == "") {
+        outcomeMessage = `It's a draw! <br> <br> Since no one wins the previous round, no one is declared the winner. <br><br> Throw again to continue the game.`;
+      }
+    }
+  }
+
+  var message = `${userName}, the computer chose ${itemName} to play against you, <br> while the computer chose ${userGuess} for you. <br> <br> ${outcomeMessage}`;
+
+  return message;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  // collect user's name as the first input after the page loads
+  if (currentGameMode == "waiting for user name") {
+    userName = input;
+    // now that we have the user's name
+    currentGameMode = "selecting game";
+    return `Hello, ${userName}! <br> <br> Please choose one of the four different games below by typing in the words in " ": <br> 1. "SPS" to play normal Scissors Paper Stone. <br> 2. "korean SPS" to play muk-jji-ppa. <br> 3. "computer SPS" to play computer vs. computer normal Scissors Paper Stone. <br> 4. "computer korean SPS" to play computer vs. computer muk-jji-ppa.`;
+  }
+
+  if (currentGameMode == "selecting game") {
+    var message = `Please enter "scissors", "paper" or "stone".`;
+    var computerMessage = `Please click the "submit" button to start the game.`;
+    if (input == "SPS") {
+      currentGameMode = "SPS";
+      return `You have selected "normal SPS game", ${userName}. <br> <br> ${message}`;
+    }
+    if (input == "korean SPS") {
+      currentGameMode = "korean SPS";
+      return `You have selected "Korean SPS game - muk-jji-ppa", ${userName}. <br> <br> ${message}`;
+    }
+    if (input == "computer SPS") {
+      currentGameMode = "computer SPS";
+      return `You have selected "computer vs computer normal SPS game", ${userName}. <br> <br> ${computerMessage}`;
+    }
+    if (input == "computer korean SPS") {
+      currentGameMode = "computer korean SPS";
+      return `You have selected "computer vs computer Korean SPS game - muk-jji-ppa", ${userName}. <br> <br> ${computerMessage}`;
+    }
+  }
+
+  if (currentGameMode == "SPS") {
+    myOutputValue = playSPS(userName, input);
+  } else if (currentGameMode == "korean SPS") {
+    myOutputValue = playKoreanSPS(userName, input);
+  } else if (currentGameMode == "computer SPS") {
+    myOutputValue = playComputerSPS(userName, input);
+  } else if (currentGameMode == "computer korean SPS") {
+    myOutputValue = playComputerKoreanSPS(userName, input);
+  }
+
```

Comment:
I like that you managed to extract the complicated game logic out into helper functions that you can call in the main function. This allows the code to be read easily